### PR TITLE
[#18833] Confirm button disappearing while editing address

### LINKED ISSUE #18833: [IOS only] The 'Confirm' button disappears in the editing address flow when the keyboard is closed
### Steps:
1. Go to the editing address flow
2. Close the keyboard
3. Check the visibility of 'Confirm' button

### Actual result:
The  'Confirm' button disappeared
image

### Expected result:
The 'Confirm' button remains visible
How it works on Android:
image

### Additional info:
The same issue applies to the editing flow of watch-only accounts.

### Device:
iPhone 11 Pro max, IOS 17

### PR DESCRIPTION
**Pull request by ulisesmac**
fixes #18833

### Summary

A small PR.

https://github.com/status-im/status-mobile/assets/90291778/7e408351-23f3-45fd-b9f3-b26ff12fcda2

This PR has also the changes made in the PR
- #19160 

Becuase they are too realted.

#### Platforms

- iOS

### Steps to test

- Open Status
- Navigate to the Wallet tab -> Pick any account card -> edit
- Now the button below is showing and disabled

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/create_account/view.cljs [hunk 1]
```diff
@@ -51,36 +51,67 @@
...
         on-change-text               #(reset! account-name %)
         primary-name                 (first (rf/sub [:contacts/contact-two-names-by-identity
                                                      public-key]))
+        show-account-origin          #(rf/dispatch
+                                       [:show-bottom-sheet {:content account-origin/view}])
         {window-width :width}        (rn/get-window)]
     (fn [{:keys [theme]}]
-      (let [{:keys [new-keypair]} (rf/sub [:wallet/create-account])]
+      (let [{:keys [new-keypair]} (rf/sub [:wallet/create-account])
+            ;;
```
Comment (smohamedjavid) [inline src/status_im/contexts/wallet/create_account/view.cljs]: Maybe a leftover

**Review by smohamedjavid — APPROVED** [review]
🚀

File: src/status_im/contexts/wallet/create_account/view.cljs [hunk 2]
```diff
@@ -51,36 +51,67 @@
...
+                                                      :derivation-path @derivation-path})}]))
+            derive-address        (fn [password]
+                                    (rf/dispatch
+                                     [:wallet/derive-address-and-add-account
+                                      {:sha3-pwd     (security/safe-unmask-data password)
+                                       :emoji        @emoji
+                                       :color        @account-color
+                                       :path         @derivation-path
+                                       :account-name @account-name}]))
+           ]
```
Comment (OmarBasem) [inline src/status_im/contexts/wallet/create_account/view.cljs]: Trailing `]`

**Review by OmarBasem — APPROVED** [review]

File: translations/en.json [hunk 3]
```diff
@@ -446,8 +446,8 @@
     "decline": "Decline",
     "decryption-failed-content": "An error occured decrypting your data. You might need to erase your old data and generate a new account. Tap “Apply” to erase or “Cancel” to try again",
     "default": "Default",
-    "default-account-placeholder": "Account {{number}}",
-    "default-watched-address-placeholder": "Watched address {{number}}",
+    "default-account-placeholder": "Account",
```
Comment (J-Son89) [inline translations/en.json]: this should be "Account Name" btw, feel free to check with @benjthayer so it will match Desktop 👍

**Review by J-Son89 — APPROVED** [review]

File: translations/en.json (hunk 3 above)
Comment (ulisesmac) [reply translations/en.json]: Oh yes, I missed the `Name` you previously mentioned. now it's fixed

File: src/status_im/contexts/wallet/create_account/view.cljs (hunk 1 above)
Comment (ulisesmac) [reply src/status_im/contexts/wallet/create_account/view.cljs]: exactly, thanks! :)

**Review by clauxx — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Hi @ulisesmac! Thanks for your PR!

I have a doubt regarding the correctness of a slide button fix.
[Here](https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1-111913&mode=design&t=frwpaHJ5LeY9erzZ-0) we have a design how it should look like with opened keyboard:
Screenshot 2024-03-18 at 11 17 58

And current fix looks like this:
Screenshot 2024-03-18 at 11 23 19

And I can not scroll here.

**Comment by ulisesmac** [comment]
Hey @mariia-skrypnyk thanks for testing.

> I have a doubt regarding the correctness of a slide button fix.
we have a design how it should look like with opened keyboard:

That's related to the floating button page implementation, we might need to revisit it, currently designs look as:

image

And our implementation:

image


Not too different, atm this is out of scope of this PR, but we can create a follow up issue.


> 
> And I can not scroll here.

You couldn't scroll because the input was focused. at the beginning I didn't think that behavior is wrong, but now I enabled it, so we are now able to scroll :+1:

**Comment by mariia-skrypnyk** [comment]
> That's related to the floating button page implementation, we might need to revisit it

Hi @ulisesmac ! Thanks for a details.
**1)** So, can I create PR for investigating the best approach for this overlapping of floating button of Origin and Derivation path blocks?
Screenshot 2024-03-20 at 09 41 50

**2)** In case 1) I still can not scroll to see the Derivation path. Can it be fixed in this PR or followup?

https://github.com/status-im/status-mobile/assets/147824399/6e6d7a9e-c4ef-41d7-b018-e916d1e234da

**Comment by ulisesmac** [comment]
> > That's related to the floating button page implementation, we might need to revisit it
> 
> Hi @ulisesmac ! Thanks for a details. **1)** So, can I create PR for investigating the best approach for this overlapping of floating button of Origin and Derivation path blocks? Screenshot 2024-03-20 at 09 41 50
> 
> **2)** In case 1) I still can not scroll to see the Derivation path. Can it be fixed in this PR or followup?
> IMG_7532.MP4

Hey @mariia-skrypnyk 

AFAIK, that's an iOS behavior. If you are focusing an input and you can scroll, it'll be limited beacuse the input should always be within the screen.

**Comment by mariia-skrypnyk** [comment]
Thanks @ulisesmac !
I'll ask @Francesca-G review as I am not sure in my opinion.
 @Francesca-G please, read our thread from https://github.com/status-im/status-mobile/pull/19186#issuecomment-2004494478 comment. Thanks!

**Comment by Francesca-G** [comment]
> Thanks @ulisesmac ! I'll ask @Francesca-G review as I am not sure in my opinion. @Francesca-G please, read our thread from [#19186 (comment)](https://github.com/status-im/status-mobile/pull/19186#issuecomment-2004494478) comment. Thanks!

Yes I think that the user should be able to scroll and see the whole screen when the keyboard is open 👍

**Comment by ulisesmac** [comment]
> > Thanks @ulisesmac ! I'll ask @Francesca-G review as I am not sure in my opinion. @Francesca-G please, read our thread from [#19186 (comment)](https://github.com/status-im/status-mobile/pull/19186#issuecomment-2004494478) comment. Thanks!
> 
> Yes I think that the user should be able to scroll and see the whole screen when the keyboard is open 👍

Hey @Francesca-G  thanks for checking this out!

I'm allowing the scroll, but I think it's an iOS issue, it seems if you have a focused input you can't scroll beyond. It's not related to the keyboard, it's because the input is being focused.
Could you please correct me if I am wrong?

In case you confirm this is actually an issue, we'd like to address it in a follow up, since rn we are focusing on solving other more important bugs.

thanks!

**Comment by Francesca-G** [comment]
> Hey @Francesca-G thanks for checking this out!
> 
> I'm allowing the scroll, but I think it's an iOS issue, it seems if you have a focused input you can't scroll beyond. It's not related to the keyboard, it's because the input is being focused. Could you please correct me if I am wrong?
> 
> In case you confirm this is actually an issue, we'd like to address it in a follow up, since rn we are focusing on solving other more important bugs.
> 
> thanks!

@ulisesmac I was experimenting with it and as you can see from the video you're able to drag the page to see the _derivation path_. The main issue is that the page stays like that after the keyboard goes down (look at the account avatar that is cut at the top of the page) and I wasn't able to drag the page back to its original position unless the keyboard was open. 

So, if we're going to address the scrolling issue this issue needs to be addressed too.


https://github.com/status-im/status-mobile/assets/122543089/4a1cc8fc-27ba-4bac-b6a2-1a6392e32156

**Comment by ulisesmac** [comment]
Hey @Francesca-G, I've fixed the issue, thanks for spotting it!

Here's a video of the new behavior:

https://github.com/status-im/status-mobile/assets/90291778/15453481-7dcf-41c8-b94b-23db85ae50e5

**Review by Francesca-G — APPROVED** [review]
looks good now, thanks @ulisesmac for fixing it ✨

**Pull request merged**